app/ide-desktop/lib/content/src/index.ts

```diff
@@ -148,75 +156,84 @@ class Main {
-                    const hideAuth = () => {
-                        const auth = document.getElementById('dashboard')
-                        const ide = document.getElementById('root')
-                        if (auth) auth.style.display = 'none'
-                        if (ide) ide.style.display = ''
-                    }
-                    /** This package is an Electron desktop app (i.e., not in the Cloud), so
-                     * we're running on the desktop. */
-                    /** TODO [NP]: https://github.com/enso-org/cloud-v2/issues/345
-                     * `content` and `dashboard` packages **MUST BE MERGED INTO ONE**. The IDE
-                     * should only have one entry point. Right now, we have two. One for the cloud
-                     * and one for the desktop. Once these are merged, we can't hardcode the
-                     * platform here, and need to detect it from the environment. */
-                    const platform = authentication.Platform.desktop
-                    /** FIXME [PB]: https://github.com/enso-org/cloud-v2/issues/366
-                     * React hooks rerender themselves multiple times. It is resulting in multiple
-                     * Enso main scene being initialized. As a temporary workaround we check whether
-                     * appInstance was already ran. Target solution should move running appInstance
-                     * where it will be called only once. */
-                    let appInstanceRan = false
-                    const onAuthenticated = () => {
-                        if (
-                            !contentConfig.OPTIONS.groups.featurePreview.options.newDashboard.value
-                        ) {
-                            hideAuth()
-                            if (!appInstanceRan) {
-                                appInstanceRan = true
-                                void appInstance.run()
-                            }
-                        }
-                    }
-                    authentication.run({
-                        logger,
-                        platform,
-                        projectManager: projectManager.ProjectManager.default(),
-                        showDashboard:
-                            contentConfig.OPTIONS.groups.featurePreview.options.newDashboard.value,
-                        onAuthenticated,
-                    })
-                } else {
-                    void appInstance.run()
-                }
                 const email = contentConfig.OPTIONS.groups.authentication.options.email.value
                 // The default value is `""`, so a truthiness check is most appropriate here.
                 if (email) {
                     logger.log(`User identified as '${email}'.`)
                 }
+                void this.app.run()
+            }
+        }
+    }
+
+    main() {
+        // Load command line arguments into `configOptions.OPTIONS`.
+        new app.App({
```

Comment:
can't we just merge configOptions.OPTIONS with host.urlParams like the App is doing in constructor rather than creating app.App and depends on leaking params into them?

Reply:
yes - but i'm not sure that's a good idea because that might get out of sync with how the `App` merges options.

result of discussion on Discord: maybe `App` could expose it as a class static method?
`app.App.mergeOptions(contentConfig.OPTIONS)` (name can be anything)